### PR TITLE
[repo] Update the System.Diagnostics.DiagnosticSource and Microsoft.Extensions.* packages to .NET 9.0 RC.1

### LINKED ISSUE #5846: Update System.Diagnostics.DiagnosticSource and Microsoft.Extensions.* packages to .NET 9.0
Update the System.Diagnostics.DiagnosticSource and Microsoft.Extensions.* packages to .NET 9.0.

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Fixes #5846

## Changes

Please provide a brief description of the changes here.
* Updated the `System.Diagnostics.DiagnosticSource` and `Microsoft.Extensions.*` packages to .NET 9.0 RC.1
* Updated `System.Text.Json` to `9.0.0-rc.1.24431.7` for tests and examples
* Updated `RELEASENOTES.md`

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: RELEASENOTES.md [hunk 1]
```diff
@@ -4,6 +4,15 @@ This file contains highlights and announcements covering all components.
 For more details see `CHANGELOG.md` files maintained in the root source
 directory of each individual package.
 
+## Unreleased
```
Comment (cijothomas) [inline RELEASENOTES.md:7]: I think this is to be updated during 1.10 update, not required in the interim beta/rc stages?

File: RELEASENOTES.md [hunk 2]
```diff
@@ -4,6 +4,15 @@ This file contains highlights and announcements covering all components.
...
+
+* Added `net9.0` targets to all the components.
+
+* Removed `net6.0` targets from all the components.
```
Comment (Kielek) [inline RELEASENOTES.md:11]: IMO worth to add more explicit note that this version is no longer supporting .net6.

**Review by cijothomas — APPROVED** [review]

File: RELEASENOTES.md (hunk 1 above)
Comment (rajkumar-rangaraj) [reply RELEASENOTES.md:7]: There was a request in the previous PR to highlight the removal of the old target and the addition of the new one. Tracking it this way ensures we won't miss updating this during the actual release. We could update the release scripts to modify it automatically.

File: RELEASENOTES.md (hunk 2 above)
Comment (rajkumar-rangaraj) [reply RELEASENOTES.md:11]: how about this one?
Removed `net6.0` targets from all components, as `.NET 6` will reach end of life on November 12, 2024.

File: RELEASENOTES.md (hunk 1 above)
Comment (CodeBlanch) [reply RELEASENOTES.md:7]: Agree with @cijothomas. This file should be a curated summary of high-level/interesting things being released in 1.10.0 (stable) which we'll add as part of that final drop. This content being added here seems too low-level/CHANGELOG-ish. If you want, you could add notes here we'll use later to make that content. Something like...

```md
<!--
## 1.10.0

TODO:
* Mention drop of net6.0 & net7.0 targets, addition of net9.0 target.
-->
```

Or you could just do nothing to this file and we can work on the release notes later 🤣

Comment (rajkumar-rangaraj) [reply RELEASENOTES.md:7]: Will remove the changes to this file to keep it simple.

**Pull request merged**